src/composables/useTresContextProvider/index.ts

```diff
@@ -18,18 +38,22 @@ export interface TresContext {
   controls: Ref<(EventDispatcher & { enabled: boolean }) | null>
   renderer: ShallowRef<WebGLRenderer>
   raycaster: ShallowRef<Raycaster>
-  perf: {
-    maxFrames: number
-    fps: {
-      value: number
-      accumulator: number[]
-    }
-    memory: {
-      currentMem: number
-      allocatedMem: number
-      accumulator: number[]
-    }
-  }
+  perf: PerformanceState
+  /**
+   * If set to 'on-demand', the scene will only be rendered when the current frame is invalidated
+   * If set to 'manual', the scene will only be rendered when advance() is called
+   * If set to 'always', the scene will be rendered every frame
+   */
+  renderMode: Ref<'always' | 'on-demand' | 'manual'>
```

Comment:
Hey, @andretchen0 I thought the same at the beginning but I think R3F and Threlte use the term `on-demand` to describe a render mode where the scene is rendered only when it's necessary, such as when changes are detected or the frame is manually invalidated. If you see it outside of the end-user persona, as authors, we are also demanding the render to be invalidated when a change is detected, since is possible to do so, one example would be:

 On the OrbitControls of `cientos`, you as author, would demand an invalidation every time the control updates.

On `manual` you need to manually render instead of invalidating the frame.

In this case, I prefer to keep the standard term from other renderers. I also want to avoid end-user confusion with vue `watch`